### DISCUSSION TITLE
How can I customise  rootToggle for multiple routes, add a folder for the routing but not appear in the sidebar

### DISCUSSION THREAD

**Discussion by spiritoftime** [fuma-nama/fumadocs#784] [Q&A] [answered]
Hello! I’m trying to customise the sidebar, but am getting stuck on how to approach it. appreciate your help =)
Code Repo: https://github.com/spiritoftime/psle-math/pull/1
so i’m trying to customise the sidebar for multiple topics. For example, right now clicking psle math takes me to [topic/psle/math/lectures](http://localhost:3000/topic/psle/math/lectures), clicking on finance takes me to topic/finance.
 ![image](https://github.com/user-attachments/assets/1f4533a5-e7fe-4d03-b175-496f17a2b637)

### Issue 1: How could I make customised root toggles based on url?

this one is likely a nextjs newb question

for example when i go to psle/math/lectures, my root toggle will have lectures and questions. this takes me to psle/math/lectures and psle/math/questions respectively. Hence, They are meant to only be for psle/math.

![image](https://github.com/user-attachments/assets/79c3cf93-6b52-4a37-ba40-f040728030da)

Now when i click to /finance, i get:

![image](https://github.com/user-attachments/assets/0ca31731-b3c9-4453-aa5a-b44f709f99f5)
I would need to make the roottoggle disappear for this site; so i would probably need to check for the route.

at layout.config:

```tsx
export const getRootToggleOptions= () => {
  // need to check route, then filter the modes

  return modes.map((mode) => ({
    url: `/topic/${mode.param}`,
    icon: (
      <mode.icon
        className="size-9 shrink-0 rounded-md bg-gradient-to-t from-fd-background/80 p-1.5"
        style={{
          backgroundColor: `hsl(var(--${mode.param}-color)/.3)`,
          color: `hsl(var(--${mode.param}-color))`,
        }}
      />
    ),
    title: mode.name,
    description: mode.description,
  }));
}
// docs layout configuration
export const docsOptions: DocsLayoutProps = {
  ...baseOptions,
  tree: pageTree,
  sidebar: {
    defaultOpenLevel: 0,
    banner: (
      <RootToggle
        options={getRootToggleOptions()}
      />
    ),
  },
};
```

where i’m stuck at is how to get the route here, I tried to add “use client” at layout.config.tsx so i could try usePathName(), then i got 

```
Module build failed: UnhandledSchemeError: Reading from "node:path" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
```

I tried to add 

```jsx
  headers.set("x-current-path", request.nextUrl.pathname);
```

to my middleware. However, I’m struggling to log out the header  at getToggleRootOptions to get the route, probably because it is not a server component 
https://www.propelauth.com/post/getting-url-in-next-server-components
Error: `headers` was called outside a request scope. Read more: https://nextjs.org/docs/messages/next-dynamic-api-wrong-context

### Issue 2: How can I get rid of the sidebar?

In /finance, I dont want PSLE and math to show up at all. It's different from https://github.com/fuma-nama/fumadocs/discussions/728, because I don't have index.mdx. All I want is to have the route be /psle/math, which is why i added the folder

At the moment, this is my mdx folder structure:

![image](https://github.com/user-attachments/assets/be100f32-bdeb-4f5d-9e0e-417fca9b15e4)

**Comment by fuma-nama** [comment]
You should create another client component and import it from the config file. In that client component, use Fumadocs Root Toggle component and customise it with options.

You may even create your own toggle component if you want more flexibility over the default one, but it's more advanced.

**Comment by fuma-nama** [comment, marked as answer]
Issue 2: Just set the finance folder to be a root folder

**Comment by spiritoftime** [comment]
thank you so much!! really was a noob question haha, appreciate your help =)
